tasks/BasePIDController.cpp

```diff
@@ -74,10 +74,26 @@ std::pair<base::LinearAngular6DCommand, LinearAngular6DPIDState> BasePIDControll
         // Angular case
         if (!base::isUnset(reference.angular(i)))
         {
-            command_out.angular[i] =
-                pid.angular[i].update(current_state.angular(i),
-                                   reference.angular(i),
-                                   current_state.time.toSeconds());
+            if (_position_in_world_frame)
+            {
+                /* calculate the angle difference as current state and set reference to zero */
+                double current_state_angular = current_state.angular(i) - reference.angular(i);
+                while(current_state_angular > M_PI)
+                    current_state_angular-=(2*M_PI);
+                while(current_state_angular < -M_PI)
+                    current_state_angular+=(2*M_PI);
+                command_out.angular[i] =
+                    pid.angular[i].update(current_state_angular,
```

Comment:
Actually, if `apply_derivative_to_error` is true,  state=Current-Target, target=0 is not equivalent to state=Current, target=Target. 

You can handle this in PIDController by doing:

~~~
double diff = base::Angle(current_state.angular(i) - reference.angular(i));
current_state.angular(i) = reference.angular(i) + diff.getRad();
~~~

Reply:
this is done, although I still had to define a variable `double current_state_angular`, since `current_state` is only passed as const ref